### DISCUSSION TITLE
setStore oddities

### DISCUSSION THREAD

**Discussion by otonashixav** [solidjs/solid#842] [Q&A] [answered]
1. The `traversed` array in the callback form of `setStore` has the traversed keys in reverse order (e.g. `setStore("a", "b", (_, t) => console.log(t)` logs `["b", "a"]`)
2. The `{ from, to, by }` method of filtering an array uses `for (let i = from; i <= to; i += by)` which includes the `to` index in contrast to methods like `Array.prototype.slice` which exclude the end index. 

#840 is as far as I can tell, a bug, but I'm wondering if these two are intentional.

**Comment by ryansolid** [comment, marked as answer]
The reverse order was intentional. I used to use a lot of methods to construct the arguments for setStore programmatically when I started. Things like making a Redux like pattern. I asked this question in the discord around 1.0. But the idea is that with unknown depth it's easier to deal with the end of the path with 0 based indexes. I thought some people would find it weird but no one really had an opinion on it either way.

Hmm.. yeah I forget where I got the to being inclusive from.. maybe it was Falcor. It was intentional at the time but maybe that's weird.
